sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java

```diff
@@ -125,7 +126,10 @@ public String toString() {
 
     @Override
     public T get() {
-      return translator.apply(value.get());
+      if (cachedValue == null) {
+        cachedValue = translator.apply(value.get());
```

Comment:
Is it ok for the translator to be called more than once, such that multiple callers may get different instances of the same translated object?